### ISSUE TITLE
打开enable_cluster_plan，update语句报错“cache lookup failed for type 0”

### ISSUE THREAD

**Issue by mass3** [ADBSQL/AntDB#39]
### 打开enable_cluster_plan，update语句报错“cache lookup failed for type 0”
```
postgres=# set enable_cluster_plan = off;
SET
postgres=# EXECUTE xc_alter_table_update(11, 'bb');
UPDATE 1
postgres=# set enable_cluster_plan = on;           
SET
postgres=# EXECUTE xc_alter_table_update(11, 'bb');
ERROR:  cache lookup failed for type 0
postgres=# 
```

测试sql;
```
CREATE TABLE xc_alter_table_3 (a int, b varchar(10)) DISTRIBUTE BY HASH(a);
INSERT INTO xc_alter_table_3 VALUES (0, NULL);
INSERT INTO xc_alter_table_3 VALUES (1, 'a');
INSERT INTO xc_alter_table_3 VALUES (2, 'aa');
INSERT INTO xc_alter_table_3 VALUES (3, 'aaa');
INSERT INTO xc_alter_table_3 VALUES (4, 'aaaa');
INSERT INTO xc_alter_table_3 VALUES (5, 'aaaaa');
INSERT INTO xc_alter_table_3 VALUES (6, 'aaaaaa');
INSERT INTO xc_alter_table_3 VALUES (7, 'aaaaaaa');
INSERT INTO xc_alter_table_3 VALUES (8, 'aaaaaaaa');
INSERT INTO xc_alter_table_3 VALUES (9, 'aaaaaaaaa');
INSERT INTO xc_alter_table_3 VALUES (10, 'aaaaaaaaaa');
-- Create some objects to check the effect of redistribution
CREATE VIEW xc_alter_table_3_v AS SELECT count(*), sum(a), avg(a) FROM xc_alter_table_3;
CREATE RULE xc_alter_table_3_insert AS ON UPDATE TO xc_alter_table_3 WHERE OLD.a = 11 DO INSERT INTO xc_alter_table_3 VALUES (OLD.a + 1, 'nnn');
PREPARE xc_alter_table_insert AS INSERT INTO xc_alter_table_3 VALUES ($1, $2);
PREPARE xc_alter_table_delete AS DELETE FROM xc_alter_table_3 WHERE a = $1;
PREPARE xc_alter_table_update AS UPDATE xc_alter_table_3 SET b = $2 WHERE a = $1;

-- Now begin the tests
ALTER TABLE xc_alter_table_3 DISTRIBUTE BY HASH(a);
SELECT count(*), sum(a), avg(a) FROM xc_alter_table_3; -- Check on tuple presence
SELECT * FROM xc_alter_table_3_v;
EXECUTE xc_alter_table_insert(11, 'b');
SELECT b FROM xc_alter_table_3 WHERE a = 11;
EXECUTE xc_alter_table_update(11, 'bb');
```

**Comment by hsiaoby** [comment]
fixed by 429b134302fe3a4f3e055bb5302f89b973fd4a5e

**Comment by mass3** [comment]
update结果正确
```
postgres=# EXECUTE xc_alter_table_insert(11, 'b');
INSERT 0 1
postgres=# SELECT b FROM xc_alter_table_3 WHERE a = 11;
 b 
---
 b
(1 row)

postgres=# EXECUTE xc_alter_table_update(11, 'bb');
UPDATE 1
postgres=# 
```
